python/ray/_private/resource_isolation_config.py

```diff
@@ -0,0 +1,249 @@
+import logging
+from typing import Optional
+
+import ray._private.ray_constants as ray_constants
+
+import ray._private.utils as utils
+
+logger = logging.getLogger(__name__)
+
+# See https://docs.kernel.org/admin-guide/cgroup-v2.html#weights
+# for information about cpu weights
+_CGROUP_CPU_MAX_WEIGHT: int = 10000
+
+
+class ResourceIsolationConfig:
+    """Configuration for enabling resource isolation by reserving memory
+    and cpu for ray system processes through cgroupv2.
+    This class validates configuration for resource isolation by
```

Comment:
nit: google style guide docstring format we usually follow is:

"""Single line description, cannot span multiple.

More details blah blah.
""""